src/main/java/com/uid2/shared/attest/AttestationTokenRetriever.java

```diff
@@ -96,16 +96,28 @@ public AttestationTokenRetriever(Vertx vertx,
         }
         this.appVersionHeader = builder.toString();
 
+        this.vertx.setPeriodic(0, attestCheckMilliseconds, this::attestationCheck);
     }
 
-    private void attestationExpirationCheck(long timerId) {
+    public void attestationCheck(long timerId) {
         // This check is to avoid the attest() function takes longer than 60sec and get called again from this method while attesting.
         if (!this.isAttesting.compareAndSet(false, true)) {
             LOGGER.warn("In the process of attesting. Skip re-attest.");
             return;
         }
 
         try {
+
+            if (!attestationProvider.isReady()) {
+                throw new AttestationTokenRetrieverException("attestation provider is not ready");
+            }
+
+            if(this.attestationToken.get() == null) {
```

Comment:
Why have you added this check and then attest? The scheduled attestationCheck should only happen after the first successful attestation

Reply:
changed back as noted above.